code/game/sound.dm

```diff
@@ -14,25 +14,30 @@
 	channel = channel || open_sound_channel()
 
  	// Looping through the player list has the added bonus of working for mobs inside containers
-	for (var/P in GLOB.player_list)
+	var/sound/S = sound(soundin)
+	var/maxdistance = (world.view + extrarange) * 3
+	for(var/P in GLOB.player_list)
 		var/mob/M = P
 		if(!M || !M.client)
 			continue
-		if(get_dist(M, turf_source) <= world.view + extrarange)
+		var/distance = get_dist(M, turf_source)
+
+		if(distance <= maxdistance)
 			var/turf/T = get_turf(M)
 			if(T && T.z == turf_source.z)
-				M.playsound_local(turf_source, soundin, vol, vary, frequency, falloff, surround, channel, pressure_affected)
+				M.playsound_local(turf_source, soundin, vol, vary, frequency, falloff, channel, pressure_affected, is_global, S)
 
-/mob/proc/playsound_local(turf/turf_source, soundin, vol as num, vary, frequency, falloff, surround = 1, channel = 0, pressure_affected = TRUE)
+/mob/proc/playsound_local(turf/turf_source, soundin, vol as num, vary, frequency, falloff, channel = 0, pressure_affected = TRUE, is_global = FALSE, sound/S)
 	if(!client || !can_hear())
 		return
 
-	soundin = get_sfx(soundin)
+	if(!S)
+		S = sound(get_sfx(soundin))
 
-	var/sound/S = sound(soundin)
 	S.wait = 0 //No queue
 	S.channel = channel || open_sound_channel()
 	S.volume = vol
+	S.environment = -1
```

Comment:
@RemieRichards 

Here is the "hacky" way to get around BYOND's bug handling of environments and why Bay's sound environment's work.If you reset it, every time, then you don't have any issues.